addon/v1alpha1/types_managedclusteraddon.go

```diff
@@ -199,6 +202,14 @@ type ConfigReference struct {
 
 	// lastObservedGeneration is the observed generation of the add-on configuration.
 	LastObservedGeneration int64 `json:"lastObservedGeneration"`
```

Comment:
Will the `LastObservedGeneration` be replaced by the `LastAppliedConfig` in the future？ should we mark it as deprecated？